rfd/0094-kubernetes-node-joining.md

```diff
@@ -0,0 +1,335 @@
+---
+authors: Hugo Hervieux (hugo.hervieux@goteleport.com)
+state: draft
+---
+
+# RFD 0094 - Kubernetes node joining
+
+## Required approvers
+
+* Engineering: @r0mant && @gus
+* Product: @klizhentas && @xinding33
+* Security: @reedloden
+
+## What
+
+This RFD discusses various methods for Teleport proxies to join a Teleport
+cluster in a same-cluster Kubernetes context and covers implementation details
+of the chosen approach.
+
+## Why
+
+We want to refactor the `teleport-cluster`  Helm chart to be a reference
+delivery method for Teleport. This involves deploying separately Teleport auth
+and proxy nodes. Proxy nodes have to register against the Teleport auth nodes
+without user interaction and without compromising the cluster security (no
+long-lived tokens).  Proxy nodes can be automatically created or deleted (by a
+Kubernetes HPA for example).
+
+User Motivation: 
+
+- As new Teleport adopter, I want to deploy a working, secure and highly
+  available Teleport cluster on Kubernetes, with a single Helm command and as
+  few values as possible so that I can start using Teleport as fast as possible.
+- As a Teleport power user, I want to easily join my custom sets of Teleport
+  nodes on Kubernetes so that I have less join automation to build and maintain.
```

Comment:
I don't know if deploying nodes as in "SSH nodes" in the same Kube cluster has much of a value tbh since we have Kube Access, I would maybe say "Teleport services like Machine ID or plugins".

Reply:
I was thinking about kubernetes_service, app_service, db_service, discovery_service, windows_desktop_service. I'll explicit the list.

Reply:
done